lib/scholar/metrics/distance.ex

```diff
@@ -13,6 +13,11 @@ defmodule Scholar.Metrics.Distance do
   D(x, y) = \\sqrt{\\sum_i (x_i - y_i)^2}
   $$
 
+  ## Options
+
+    * `:axes` - Axes to aggregate distance over. If `:axes` set to `nil` then function does not aggregate distances.
+      Defaults to `nil`.
```

Comment:
Perhaps something like this?

```suggestion
    * `:axes` - Axes to calculate the distance over. By default the distance
      is calculated between the whole tensors.
```